roofit/roofitcore/src/RooHistFunc.cxx

```diff
@@ -194,6 +194,10 @@ double RooHistFunc::evaluate() const
   return ret ;
 }
 
+void RooHistFunc::translate(RooFit::Detail::CodeSquashContext &ctx) const
+{
+   RooHistPdf::rooHistTranslateImpl(this, ctx, _intOrder, _dataHist, _histObsList);
```

Comment:
```suggestion
   RooHistPdf::rooHistTranslateImpl(this, ctx, _intOrder, _dataHist, _depList);
```
You have to use the list of observables in the function (the ones in the histogram might have different names, and then the context doesn't handle it correctly)

Reply:
They actually did, that's why I added the internal var code path in the calculate tree index function.